aten/src/ATen/native/Activation.cpp

```diff
@@ -194,21 +198,64 @@ Tensor & celu_(Tensor & self, Scalar alpha) {
   return at::elu_(self, alpha, Scalar(1.0), Scalar(inv_alpha));
 }
 
+std::tuple<Tensor, Tensor> silu_forward(const Tensor& input) {
+  Tensor result = at::native::empty_like(input, at::MemoryFormat::Contiguous);
+  Tensor buffer = at::native::empty_like(input, at::MemoryFormat::Contiguous);
+  at::silu_forward_out(result, buffer, input);
+  return std::forward_as_tuple(result, buffer);
+}
+
+std::tuple<Tensor&, Tensor&> silu_forward_out(
+    Tensor& result,
+    Tensor& buffer,
+    const Tensor& input) {
+  auto iter = TensorIteratorConfig()
+                  .set_check_mem_overlap(true)
+                  .add_output(result)
+                  .add_output(buffer)
+                  .add_input(input)
+                  .build();
+  silu_stub(iter.device_type(), iter);
+  return std::forward_as_tuple(result, buffer);
```

Comment:
at least on cuda, since you have to read `x` in backward anyway, recomputing `sigmoid` is pretty cheap (compared to writing it in forward and reading it in backward), so given that it also requires extra memory, maybe you can rewrite it without buffer?

Reply:
Yes, this is a good point, actually I am testing this and trying to find a proper way to do that on CUDA.

Reply:
From my testing, removing the buffer will make cpu backward slightly slower but forward much faster. For CUDA side, both forward and backward become faster. I have updated the code as well as the benchmark results.